lib/docker/artifact_image.js

```diff
@@ -15,6 +16,21 @@ import sleep from '../util/sleep';
 
 let debug = Debug('docker-worker:artifactImage');
 
+async function decompressLz4File(inputFile) {
+  let outputFile = path.join(path.dirname(inputFile), path.basename(inputFile, '.lz4'));
+  // decompress using lz4
+  let input = fs.createReadStream(inputFile);
+  let output = fs.createWriteStream(outputFile);
+  let decoder = lz4.createDecoderStream();
+  input.pipe(decoder).pipe(output);
+
+  await new Promise((accept, reject) => {
+    decoder.on('end', accept);
+    decoder.on('error', reject);
+  });
```

Comment:
This theoretically creates a potential deadlock. Imagine after the call to `pipe` it immediately raises, say, the `error` event, before `decoder.on('error', reject)` is called, then the promise would never be rejected. Although in practice this is really hard to happen, if not impossible, I suggest changing it to this:

```javascript
let promise = new Promise((accept, reject) => {
  decoder.on('end', accept);
  decoder.on('error', reject);
});
input.pipe(decoder).pipe(output);
await promise;
```